derive-topn-from-window.md

```diff
@@ -0,0 +1,204 @@
+|           └─TableFullScan_12     | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo                                                                   |
++----------------------------------+----------+-----------+---------------+--------------------------------------------------------------------------------------------------+
+```
+
+In this query, the SQL is not rewritten because the partition column is not a prefix of the primary key.
+
+#### Example 6: PARTITION BY column is a prefix of the primary key but not a clustered index
+
+```sql
+CREATE TABLE t(id1 int, id2 int, value1 int, value2 int, primary key(id1,id2) nonclustered);
+SET tidb_opt_derive_topn=on;
+EXPLAIN SELECT * FROM (SELECT ROW_NUMBER() OVER (PARTITION BY id1) AS rownumber FROM t use index()) dt WHERE rownumber <= 3;
+```
+
+The result is as follows:
+
+```
++----------------------------------+----------+-----------+---------------+-----------------------------------------------------------------------------------------------+
+| id                               | estRows  | task      | access object | operator info                                                                                 |
++----------------------------------+----------+-----------+---------------+-----------------------------------------------------------------------------------------------+
+| Projection_9                     | 8000.00  | root      |               | Column#7                                                                                      |
+| └─Selection_10                   | 8000.00  | root      |               | le(Column#7, 3)                                                                               |
+|   └─Shuffle_15                   | 10000.00 | root      |               | execution info: concurrency:5, data sources:[TableReader_13]                                  |
+|     └─Window_11                  | 10000.00 | root      |               | row_number()->Column#7 over(partition by test.t.id1 rows between current row and current row) |
+|       └─Sort_14                  | 10000.00 | root      |               | test.t.id1                                                                                    |
+|         └─TableReader_13         | 10000.00 | root      |               | data:TableFullScan_12                                                                         |
+|           └─TableFullScan_12     | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo                                                                |
++----------------------------------+----------+-----------+---------------+-----------------------------------------------------------------------------------------------+
+```
+
+In this query, although the PARTITION BY column is a prefix of the primary key, the SQL is not rewritten because the primary key is not a clustered index.
```

Comment:
```suggestion
In this query, although the `PARTITION BY` column is a prefix of the primary key, the SQL is not rewritten because the primary key is not a clustered index.
```